lib/middleware/bearerToken.ts

```diff
@@ -0,0 +1,60 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import jwt from "jsonwebtoken";
+import dbConnector from "@lib/integration/dbConnector";
+import executeQuery from "@lib/integration/queryManager";
+import { QueryResult } from "pg";
+
+export interface BearerTokenPayload {
+  formID?: string;
+}
+
+const validate = (
+  handler: (req: NextApiRequest, res: NextApiResponse, options?: unknown) => void
+) => {
+  return async (req: NextApiRequest, res: NextApiResponse): Promise<unknown> => {
+    try {
+      const token = getBearerToken(req);
+      const bearerTokenPayload = jwt.verify(token, process.env.TOKEN_SECRET || "");
+      return handler(req, res, bearerTokenPayload);
+    } catch (err) {
+      res.status(403).json({ error: "Missing or invalid bearer token." });
+    }
+  };
+};
+
+const getBearerToken = (req: NextApiRequest) => {
+  const authHeader = String(req.headers["authorization"] || "");
+  if (authHeader.startsWith("Bearer ")) {
+    return authHeader.substring(7, authHeader.length);
+  } else {
+    throw new Error("Missing bearer token.");
+  }
+};
+
+export const createTemporaryToken = (email: string): string => {
```

Comment:
I started putting it in there, but there aren't any other database functions in `crud.ts`, so it didn't really make much sense.

Reply:
Hmmm doesn't really make sense here as well. Considering this is a crud operation thought it would make more sense in that file. I think you can move this then to the handler file itself. If it needs to be used by something else we can figure out a place for it